gravitee-am-service/src/main/java/io/gravitee/am/service/impl/PasswordPolicyServiceImpl.java

```diff
@@ -139,6 +145,34 @@ public Single<PasswordPolicy> update(ReferenceType referenceType, String referen
                         .type(EventType.PASSWORD_POLICY_UPDATED).throwable(error)));
     }
 
+    @Override
+    public Completable delete(ReferenceType referenceType, String referenceId, String policyId, User principal) {
+        log.debug("Delete password policy with id '{}' for {} {}", policyId, referenceType, referenceId);
+        return passwordPolicyRepository.findByReferenceAndId(referenceType, referenceId, policyId)
+                .flatMapSingle(policy -> resetPolicyOnIdentityProviders(referenceType, referenceId, policyId)
+                        .andThen(Completable.defer(() -> passwordPolicyRepository.delete(policy.getId())))
```

Comment:
```suggestion
                        .andThen(passwordPolicyRepository.delete(policy.getId()))
```

why do you put Defer here?

Reply:
It prevents deleting password policy if there ware no update in Identity providers, on call: `resetPolicyOnIdentityProviders()`